scripts/workers/code_upload_submission_worker.py

```diff
@@ -127,14 +203,121 @@ def create_job_object(message, environment_image):
+    # Used to create submission file by phase_pk and selecting dataset location
+    submission_pk = message["submission_pk"]
+    challenge_pk = message["challenge_pk"]
+    phase_pk = message["phase_pk"]
+    SUBMISSION_PK_ENV = client.V1EnvVar(
+        name="SUBMISSION_PK", value=str(submission_pk)
+    )
+    CHALLENGE_PK_ENV = client.V1EnvVar(
+        name="CHALLENGE_PK", value=str(challenge_pk)
+    )
+    PHASE_PK_ENV = client.V1EnvVar(name="PHASE_PK", value=str(phase_pk))
+    # Using Default value 1 day = 86400s as Time Limit.
+    SUBMISSION_TIME_LIMIT_ENV = client.V1EnvVar(
+        name="SUBMISSION_TIME_LIMIT",
+        value=str(message["submission_time_limit"]),
+    )
+    SUBMISSION_TIME_DELTA_ENV = client.V1EnvVar(
+        name="SUBMISSION_TIME_DELTA", value="3600"
+    )
+    AUTH_TOKEN_ENV = client.V1EnvVar(name="AUTH_TOKEN", value=AUTH_TOKEN)
+    EVALAI_API_SERVER_ENV = client.V1EnvVar(
+        name="EVALAI_API_SERVER", value=EVALAI_API_SERVER
+    )
+    image = message["submitted_image_uri"]
+    # Get init container
+    init_container = get_init_container(submission_pk)
+    # Get dataset volume and volume mounts
+    volume_mount_list = get_volume_mount_list("/dataset", True)
+    volume_list = get_volume_list()
+    # Create and Mount Script Volume
+    script_config_map_name = "evalai-scripts-cm"
+    create_script_config_map(script_config_map_name)
+    script_volume_name = "evalai-scripts"
+    script_volume = get_config_map_volume_object(
+        script_config_map_name, script_volume_name
+    )
+    volume_list.append(script_volume)
+    script_volume_mount = get_volume_mount_object(
+        "/evalai_scripts", script_volume_name, True
+    )
+    volume_mount_list.append(script_volume_mount)
+    # Create and Mount submission Volume
+    submission_path = "/submission"
+    SUBMISSION_PATH_ENV = client.V1EnvVar(
+        name="SUBMISSION_PATH", value=submission_path
+    )
+    submission_volume_name = "submissions-dir"
+    submission_volume = get_empty_volume_object(submission_volume_name)
+    volume_list.append(submission_volume)
+    submission_volume_mount = get_volume_mount_object(
+        submission_path, submission_volume_name
+    )
+    volume_mount_list.append(submission_volume_mount)
+    # Configure Pod submission container
+    submission_container = client.V1Container(
+        name="submission",
+        image=image,
+        env=[
+            PYTHONUNBUFFERED_ENV,
+            SUBMISSION_PATH_ENV,
+            CHALLENGE_PK_ENV,
+            PHASE_PK_ENV,
+        ],
+        resources=client.V1ResourceRequirements(
+            limits={"nvidia.com/gpu": "1"}
         ),
-        spec=spec,
+        volume_mounts=volume_mount_list,
+    )
+    # Configure Pod sidecar container
+    sidecar_container = client.V1Container(
+        name="sidecar-container",
+        image="busybox",
+        command="/bin/sh",
+        args=["-c", "sh /evalai_scripts/monitor_submission.sh"],
```

Comment:
@savish28 this won't work in a `busybox` container. We have to use the ubuntu image and install curl explicitly. See my PR for reference. Also, the args might also not work. I had to pass the args as part of the command.

Reply:
Okay, updated.